include/swift/AST/DiagnosticsSema.def

```diff
@@ -2651,6 +2651,17 @@ ERROR(override_method_different_generic_sig,none,
       "base method's generic signature %2; expected generic signature to be %3",
       (DeclBaseName, StringRef, StringRef, StringRef))
 
+ERROR(requires_super_attr_not_valid_final_method, none, 
+      "'@requiresSuper' cannot be applied to final method", ())
+ERROR(requires_super_attr_only_valid_on_class_method, none, 
+      "'@requiresSuper' can only be applied on non-final class methods", ())
```

Comment:
Nit: `non-final class methods` means this to my mind

```
class Foo {
  @requiresSuper class func bar() {}
}
```

Perhaps 

```suggestion
      "'@requiresSuper' can only be applied on non-final methods on classes"
```

Or

```suggestion
      "'@requiresSuper' on method 'foo' of protocol 'Bar' is not allowed; only class methods may require calls to 'super'"
```

Reply:
This is a good suggestion! I can see why it would be confusing. I’ll make the amendment, thank you!